src/gui_common/SpeciesDetailsPanel.cs

```diff
@@ -0,0 +1,93 @@
+﻿using System;
+using Godot;
+
+public class SpeciesDetailsPanel : PanelContainer
+{
+    [Export]
+    public NodePath SpeciesDetailsLabelPath = null!;
+
+    [Export]
+    public NodePath SpeciesPreviewPath = null!;
+
+    [Export]
+    public NodePath HexPreviewPath = null!;
+
+    [Export]
+    public NodePath FossilisationButtonPath = null!;
+
+    [Export]
+    public NodePath FossilisationDialogPath = null!;
+
+    private CustomRichTextLabel? speciesDetailsLabel;
+    private SpeciesPreview? speciesPreview;
+    private CellHexesPreview? hexesPreview;
+    private Button? fossilisationButton;
+    private FossilisationDialog fossilisationDialog = null!;
+
+    private Species? previewSpecies;
+
+    public Species? PreviewSpecies
+    {
+        get => previewSpecies;
+        set
+        {
+            if (previewSpecies == value)
+                return;
+
+            previewSpecies = value;
+
+            if (previewSpecies != null)
+                UpdateSpeciesPreview();
+        }
+    }
+
+    public override void _Ready()
+    {
+        base._Ready();
+
+        speciesDetailsLabel = GetNode<CustomRichTextLabel>(SpeciesDetailsLabelPath);
+        speciesPreview = GetNode<SpeciesPreview>(SpeciesPreviewPath);
+        hexesPreview = GetNode<CellHexesPreview>(HexPreviewPath);
+        fossilisationButton = GetNode<Button>(FossilisationButtonPath);
+        fossilisationDialog = GetNode<FossilisationDialog>(FossilisationDialogPath);
+
+        if (previewSpecies != null)
+            UpdateSpeciesPreview();
+    }
+
+    /// <summary>
+    ///   Updates displayed species information based on the set preview species.
+    /// </summary>
+    private void UpdateSpeciesPreview()
+    {
+        if (speciesPreview == null || hexesPreview == null || fossilisationButton == null ||
```

Comment:
You could just check one variable. (And I think the 'ready' approach is used commonly elsewhere?)

Reply:
I had the same thought but didn't ask for that as I think I've checked up to two variables like this... so checking 4 seems pretty minor, but I think just making one of the fields nullable would be a slight code improvement here.

Reply:
Do you have a preference for which one field should be nullable then?

Reply:
I'd personally just pick the first of these fields.